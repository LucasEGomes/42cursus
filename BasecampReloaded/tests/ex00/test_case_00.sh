rm -rf temp test_case_00.user
mkdir -p temp
tar -xf $ROOT_DIR/ex00/exo.tar -C temp/
RESULT="$(ls -l temp/)"
while read p;
do
    echo "$RESULT" | grep -E -e "$p" >> test_case_00.user
done < "test_case_00.expected"
diff test_case_00.user <(echo "$RESULT") > test_case_00.output
if [ -z $(cat test_case_00.output) ];
then
    echo -n "O" >> $TEST_SUMMARY
else
    echo -n "X" >> $TEST_SUMMARY
fi
