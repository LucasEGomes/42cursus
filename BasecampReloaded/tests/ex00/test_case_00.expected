total *[0-9]{2}
drwx--xr-x *2 *[a-z][-a-z0-9]* *[a-z][-a-z0-9]* *[0-9]* *Jun *1 *(20:47|[0-9]{4}) *test0
-rwx--xr-- *1 *[a-z][-a-z0-9]* *[a-z][-a-z0-9]* *4 *Jun *1 *(21:46|[0-9]{4}) *test1
dr-x---r-- *2 *[a-z][-a-z0-9]* *[a-z][-a-z0-9]* *[0-9]* *Jun  1 *(22:45|[0-9]{4}) *test2
-r-----r-- *2 *[a-z][-a-z0-9]* *[a-z][-a-z0-9]* *1 *Jun *1 *(23:44|[0-9]{4}) *test3
-rw-r----x *1 *[a-z][-a-z0-9]* *[a-z][-a-z0-9]* *2 *Jun *1 *(23:43|[0-9]{4}) *test4
-r-----r-- *2 *[a-z][-a-z0-9]* *[a-z][-a-z0-9]* *1 *Jun *1 *(23:44|[0-9]{4}) *test5
l[-rwx]{9} *1 *[a-z][-a-z0-9]* *[a-z][-a-z0-9]* *5 *Jun *1 *(22:20|[0-9]{4}) *test6 *-> *test0
